src/main.rs

```diff
@@ -342,3 +307,75 @@ fn err_to_warp(e: podsync::Error) -> impl warp::Reply {
 pub struct QuerySince {
     since: crate::time::Timestamp,
 }
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+enum UsernameFormat {
+    Name,
+    NameJson,
+}
+impl UsernameFormat {
+    pub fn convert<'a>(&self, username: &'a str) -> Result<&'a str, warp::Rejection> {
+        match self {
+            Self::Name => Ok(username),
+            Self::NameJson => split_format_json(username).map_err(warp::reject::custom),
```

Comment:
Why would reject::custom be a 500 error in this case? Doesn't it just wrap the http::StatusCode::BAD_REQUEST (400) returned from 'split_format_json'?

Reply:
Originally I think I thought the same, but warp requires that we add a [`recover` filter](https://docs.rs/warp/latest/warp/trait.Filter.html#method.recover) to convert our error to something it can send back, and if there isn't a `recover`, it'll throw out a 500.

This would work for most of the endpoints, but in some ([like below](https://github.com/bobrippling/podsync/pull/2#discussion_r1155094656)) I'd like to short-circuit to a 4xx. I can include this in a general error handling change later on though